### PR TITLE
Add `IconButton.filled`, `IconButton.filledTonal`, `IconButton.outlined`

### LINKED ISSUE #111800: Add factory constructors for filled, tonal, outlined icon buttons
# Proposal
`IconButton.filled`, `IconButton.outlined`, etc.

Relevant: #103528, #104357

### PR DESCRIPTION
**Pull request by QuncCccccc**
Fixes #111800

This PR is to add IconButton.filled, IconButton.tilledTonal and IconButton.outlined constructors.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/lib/material/icon_button/icon_button.3.dart [hunk 1]
```diff
@@ -36,153 +36,105 @@ class DemoIconToggleButtons extends StatefulWidget {
...
+              isSelected: standardSelected,
+              icon: const Icon(Icons.settings_outlined),
+              selectedIcon: const Icon(Icons.settings),
+              onPressed: () {
+                setState(() {
+                  standardSelected = !standardSelected;
+                });
+              },
+            ),
+            const SizedBox(width: 10,),
```
Comment (HansMuller) [inline examples/api/lib/material/icon_button/icon_button.3.dart:62]: In cases like this we don't usually include the trailing comma:
```suggestion
            const SizedBox(width: 10),
```

Here and elsewhere.

File: packages/flutter/lib/src/material/icon_button.dart [hunk 2]
```diff
@@ -825,7 +939,16 @@ class _IconButtonM3 extends ButtonStyleButton {
   /// * `splashFactory` - Theme.splashFactory
   @override
   ButtonStyle defaultStyleOf(BuildContext context) {
-    return _IconButtonDefaultsM3(context);
+    switch (variant) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/icon_button.dart:942]: NICE

**Review by HansMuller — APPROVED** [review]
Nicely done. LGTM

**Comment by rydmike** [comment]
@QuncCccccc and @HansMuller, this PR is very nice, love it! However, and sadly based on my tests, it brings the same challenge to **theming** the `IconButton` as we already have with the `FilledButton` [here](https://github.com/flutter/flutter/issues/118063). If we theme the buttons `foreground` or `background` colors, we destroy the variants, as they all become the same, since they share the same theme.

A solution to this, with variant themes, is discussed in the related FAB issue [here](https://github.com/flutter/flutter/issues/107946#issuecomment-1193422262)

**Pull request merged**